packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -204,20 +204,28 @@ distributionUrl=https\\://services.gradle.org/distributions/gradle-$gradleVersio
 /// The Android plugin version is specified in the [build.gradle] file within
 /// the project's Android directory.
 String getGradleVersionForAndroidPlugin(Directory directory, Logger logger) {
-  final File buildFile = directory.childFile('build.gradle');
+  const String buildFileName = 'build.gradle/build.gradle.kts';
```

Comment:
I think either this variable should be removed and the logger strings below should have this inline or the logger strings should stay the same using the file object and rely on the if statement below to use set `buildfile` correctly

Reply:
I went for a variable because that string occured in 3 `logger.printTrace()` statements and remembered [rule-of-3](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)).


I'll go with this:

> the logger strings should stay the same using the file object and rely on the if statement below to use set buildfile correctly

I would also add `logger.error()` so that users who (for whatever weird reason) don't have a `build.gradle`/`build.gradle.kts` file will see red text when `flutter build`ing.

Reply:
I wrote some of this code originally and I it is more likely that we have a bug in our code than that the users code does not match the conditions below. 

Specifically the regular expressions assume a groovy style syntax. Can you file an issue and put it in a comment here that this code needs to be updated with regular expressions and tests that cover the kotlin case in addition to groovy?

Reply:
I filed https://github.com/flutter/flutter/issues/141410 can you add a reference to that bug in this method?

Reply:
If you mean specifically the [`_buildAndroidGradlePluginRegExp`](https://github.com/flutter/flutter/blob/ef3de359de32adad829252d7ad148d81cdfbe6be/packages/flutter_tools/lib/src/android/gradle_utils.dart#L82-L83) RegExp that is used in this method (`getGradleVersionForAndroidPlugin()`), then it works in both Groovy and Kotlin.